main.tf

```diff
@@ -193,7 +198,7 @@ resource "time_sleep" "wait_for_authorization_policy" {
 
 resource "ibm_is_vpc_routing_table" "route_table" {
   for_each                      = module.dynamic_values.routing_table_map
-  name                          = var.prefix != null ? "${var.prefix}-${var.name}-route-${each.value.name}" : "${var.name}-route-${each.value.name}"
+  name                          = var.prefix != null ? "${var.prefix}-${var.name}-route-${each.value.name}" : "${var.routing_table_name}-${each.value.name}"
```

Comment:
I don't see any reason to add `var.routing_table_name` - you can achieve this naming with the current code using `var.name`. There can be multiple routing tables here, so we will always be appending `${each.value.name}` to the name here

Reply:
As the objective of the task is to give freedom to the user to name it anything they like, I have added these as variables. If they don't provide the value then it will pick up either prefix or name.

Reply:
But how are you going to handle the naming of multiple routing tables with 1 string variable. You dont want them all named the same, so hence you will need to use `${each.value.name}` in the naming